### PR TITLE
doc: Review basic mapping

### PR DESCRIPTION
**Pull request by GromNaN**
|      Q       |   A
|------------- | -----------
| Type         | doc
| BC Break     | no
| Fixed issues | -

### CODE REVIEW

File: docs/en/reference/annotations-reference.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ Change the metadata driver configuration to use the ``AttributeDriver``:
     - $config->setMetadataDriverImpl(AnnotationsDriver::create(__DIR__ . '/Documents'));
     + $config->setMetadataDriverImpl(AttributeDriver::create(__DIR__ . '/Documents'));
 
-Replace the `@ORM\Document` annotations with the `#[ORM\Document]` attribute.
+Replace the ``@ORM\Document`` annotations with the ``#[ORM\Document]`` attribute.
```
Comment (GromNaN) [inline docs/en/reference/annotations-reference.rst:23]: double backticks are required in rst for code.

File: docs/en/reference/basic-mapping.rst [hunk 2]
```diff
@@ -11,22 +11,18 @@ Mapping Drivers
 Doctrine provides several different ways for specifying object
 document mapping metadata:
 
--  Attributes
--  XML
+-  `Attributes <annotations-reference>`_
+-  `XML <xml-mapping>`_
```
Comment (GromNaN) [inline docs/en/reference/basic-mapping.rst:15]: I find is useful to have links to more details.

File: docs/en/reference/basic-mapping.rst [hunk 3]
```diff
@@ -11,22 +11,18 @@ Mapping Drivers
...
-    If you're wondering which mapping driver gives the best
-    performance, the answer is: None. Once the metadata of a class has
-    been read from the source (attributes or xml) it is stored
-    in an instance of the
-    ``Doctrine\ODM\MongoDB\Mapping\ClassMetadata`` class and these
-    instances are stored in the metadata cache. Therefore at the end of
-    the day all drivers perform equally well. If you're not using a
-    metadata cache (not recommended!) then the XML driver might have a
-    slight edge in performance due to the powerful native XML support
-    in PHP.
```
Comment (GromNaN) [inline docs/en/reference/basic-mapping.rst:29]: Using attributes may be faster than XML now. Is was not the case for annotations. Unless we do a benchmark, this is only hypothesis. So I remove this sentence from the doc.

File: docs/en/reference/basic-mapping.rst [hunk 4]
```diff
@@ -61,6 +57,8 @@ to be designated as a document. This can be done through the
 
         namespace Documents;
 
+        use Doctrine\ODM\MongoDB\Mapping\Annotations\Document;
```
Comment (GromNaN) [inline docs/en/reference/basic-mapping.rst:60]: This page is for newcomers. Having the use statements is useful.

File: docs/en/reference/basic-mapping.rst [hunk 5]
```diff
@@ -280,16 +283,16 @@ Here is an example how to manually set a string identifier for your documents:
...
+        use Doctrine\ODM\MongoDB\Mapping\Annotations\Id;
+
         #[Document]
         class MyPersistentClass
         {
             #[Id(strategy: 'NONE', type: 'string')]
-            private string $id;
-
-            public function setId(string $id): void
```
Comment (GromNaN) [inline docs/en/reference/basic-mapping.rst:289]: No need for the setter with a public property.

File: docs/en/reference/xml-mapping.rst [hunk 6]
```diff
@@ -25,13 +25,6 @@ trunk.
 
     </doctrine-mongo-mapping>
 
-.. note::
-
-    If you do not want to use latest XML Schema document please use link like
-    `http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping-1.0.0-BETA12.xsd <http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping-1.0.0-BETA12.xsd>`_.
```
Comment (GromNaN) [inline docs/en/reference/xml-mapping.rst:31]: This file does not exist: https://github.com/doctrine/doctrine-website-build-prod/tree/master/schemas/odm

**Review by alcaeus — APPROVED** [review]

**Pull request merged**